src/services/server/auth/supabase-auth.ts

```diff
@@ -27,23 +31,25 @@ export const SupabaseAuth = inject(
     constructor(
       private createSupabaseClient: CreateSupabaseClient,
       private userRepository: UserRepository,
+      private invitationsRepository: InvitationsRepository,
+      private cookies: ICookies,
     ) {}
 
     async signUpWithEmailAndSendOTP(
       email: string,
       name: string,
       avatar: Avatar,
-      type: UserType,
     ): Promise<void> {
       const supabase = this.createSupabaseClient();
+      const invitedBy = await this.loadInvitedByFromCookies();
 
-      const { error } = await supabase.auth.admin.createUser({
+      const { data, error } = await supabase.auth.admin.createUser({
         email,
         email_confirm: true,
         user_metadata: {
           name,
           avatar,
-          type,
+          type: invitedBy ? UserType.Player : UserType.Challenger,
```

Comment:
No, because users will only ever start as either a player or a challenger. They can become a hybrid user as a challenger if they click an invitation link after already signing up, and can become a hybrid user as a player by clicking "take the challenge," but they will never sign up as one